### PR TITLE
Support only extension in war

### PR DESCRIPTION
**Pull request by arjantijms**
Support war files that only have an extension and no beans.

Fixes TCK tests:

* ChangeObserverQualifierTest
* CustomStereotypeTest
* SyntheticBeanInjectionPointTest

Refactoring:

- Generics
- Formatting
- Abbreviations
- if/else ordering

Signed-off-by: Arjan Tijms <arjan.tijms@gmail.com>

### CODE REVIEW

File: nucleus/common/internal-api/src/main/java/org/glassfish/internal/data/EngineInfo.java [hunk 1]
```diff
@@ -100,29 +101,35 @@ public void clean(ExtendedDeploymentContext context) throws Exception {
...
-        if (logger.isLoggable(Level.FINE)) {
+
+        if (logger.isLoggable(FINE)) {
             logger.fine("Container " + getContainer().getName() + " stopped");
         }
     }
+
+    @Override
+    public String toString() {
+        return (deployer != null? deployer.toString() : "")  + super.toString();
```
Comment (dmatej) [inline nucleus/common/internal-api/src/main/java/org/glassfish/internal/data/EngineInfo.java:133]: just replace it with `return deployer + ", " + super.toString();`
Btw super.toString is Object.toString(), is it what is desired?

**Review by dmatej — COMMENTED** [review]
Except those comments and Jenkins failure it seems fine to me.

**Comment by dmatej** [comment]
... but if Jenkins says that it is broken ... (beans.xml?)

File: appserver/web/gf-weld-connector/src/main/java/org/glassfish/weld/connector/WeldUtils.java [hunk 2]
```diff
@@ -521,7 +551,8 @@ public static String getBeanDiscoveryMode(InputStream beansXmlInputStream) {
         }
 
         if (beanDiscoveryMode == null) {
-            return "all";
```
[the remark this answers is not in the document]
Comment (arjantijms) [reply appserver/web/gf-weld-connector/src/main/java/org/glassfish/weld/connector/WeldUtils.java:524]: Sorry, there was refactoring first but also changes in the PR. That one should have gone into the second commit, but alas.

**Comment by arjantijms** [comment]
> ... but if Jenkins says that it is broken ... (beans.xml?)

Quite peculiar, basic authentication (non CDI) fails:

```
FAILED: testAuthGroupMappedUser
java.lang.AssertionError: BASIC auth: Group mapped user, testuser42 expected [false] but found [true]
```

**Review by dmatej — APPROVED** [review]

**Pull request merged**